client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-actions/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { Gridicon } from '@automattic/components';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useState, useRef } from 'react';
+import PopoverMenu from 'calypso/components/popover-menu';
+import PopoverMenuItem from 'calypso/components/popover-menu/item';
+import type { SiteNode } from '../types';
+import type { ReactElement } from 'react';
+
+import './style.scss';
+
+interface Props {
+	isLargeScreen?: boolean;
+	site: SiteNode;
+}
+
+export default function SiteActions( { isLargeScreen, site }: Props ): ReactElement {
+	const translate = useTranslate();
+
+	const [ isOpen, setIsOpen ] = useState( false );
+
+	const buttonActionRef = useRef< HTMLDivElement | null >( null );
+
+	const showActions = () => {
+		setIsOpen( true );
+	};
+
+	const closeDropdown = () => {
+		setIsOpen( false );
+	};
+
+	const siteUrl = site?.value?.url;
+	const siteError = site?.error;
+	const siteId = site?.value?.blog_id;
+
+	const handleClickMenuItem = () => {
+		// Handle track event here
+	};
+
+	return (
+		<>
+			<div
+				onClick={ showActions }
+				onKeyPress={ showActions }
+				role="button"
+				tabIndex={ 0 }
+				ref={ buttonActionRef }
+				className={ classNames(
+					isLargeScreen
+						? 'site-actions__actions-large-screen'
+						: 'site-actions__actions-small-screen'
+				) }
+			>
+				<Gridicon icon="ellipsis" size={ 18 } className="site-actions__all-actions" />
+			</div>
```

Comment:
Why not use semantic markup by using the actual `Button` component from `'@automattic/components'` with `borderless` and `compact` props?
That way, we could get rid of `onKeyPress`, `role` and `tabIndex` props and have better a11y.